### PR TITLE
py-tables: add v3.8.0

### PR DESCRIPTION
**Pull request by adamjstewart**
Trying to debug #40223 but this latest version also has issues. Going to open an issue upstream.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Note that I'm unable to test this, so this PR shouldn't be merged unless someone else can confirm it builds correctly.

**Review by tldahlgren — APPROVED** [review]
Confirmed homepages and (new) version sha256.

File: var/spack/repos/builtin/packages/py-tables/package.py [hunk 1]
```diff
@@ -25,26 +26,35 @@ class PyTables(PythonPackage):
...
     depends_on("py-numexpr@2.6.2:", type=("build", "run"))
-    depends_on("py-six@1.9.0:", when="@:3.5", type=("build", "run"))
-    depends_on("py-packaging", when="@3.7.0:", type=("build", "run"))
+    depends_on("py-blosc2@2.0", type=("build", "run"))
+    depends_on("py-packaging", when="@3.7:", type=("build", "run"))
+    depends_on("py-py-cpuinfo", when="@3.8:", type=("build", "run"))
+
     # tables/req_versions.py
-    depends_on("hdf5@1.8.4:1.8", when="@:3.3")
     depends_on("hdf5@1.8.4:", when="@3.4.0:")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-tables/package.py:47]:
```
depends_on("hdf5@1.10.5:", when="@3.8.0:")
```
see [req_versions.py#12](https://github.com/PyTables/PyTables/blob/v3.8.0/tables/req_versions.py#L12)

**Review by manuelakuhn — COMMENTED** [review]
Also `py-tables` does not build for me with the error
```
     15      Preparing metadata (pyproject.toml): started
     16      Running command Preparing metadata (pyproject.toml)
     17      /tmp/lzo_version_daten6_8cnwo.c: In function 'main':
     18      /tmp/lzo_version_daten6_8cnwo.c:2:5: warning: implicit declaration of function 'lzo_version_date' [-W
           implicit-function-declaration]
     19          2 |     lzo_version_date();
     20            |     ^~~~~~~~~~~~~~~~
  >> 21    /usr/bin/ld: cannot find -llzo2: No such file or directory
  >> 22      collect2: error: ld returned 1 exit status
     23      /tmp/lzo_version_date7u3iiqnr.c: In function 'main':
     24      /tmp/lzo_version_date7u3iiqnr.c:2:5: warning: implicit declaration of function 'lzo_version_date' [-W
           implicit-function-declaration]
     25          2 |     lzo_version_date();
     26            |     ^~~~~~~~~~~~~~~~
  >> 27      /usr/bin/ld: cannot find -llzo: No such file or directory
  >> 28      collect2: error: ld returned 1 exit status
     29      /tmp/BZ2_bzlibVersionh4rtwokt.c: In function 'main':
     30      /tmp/BZ2_bzlibVersionh4rtwokt.c:2:5: warning: implicit declaration of function 'BZ2_bzlibVersion' [-W
           implicit-function-declaration]
     31          2 |     BZ2_bzlibVersion();
     32            |     ^~~~~~~~~~~~~~~~
  >> 33      /usr/bin/ld: cannot find -lbz2: No such file or directory
  >> 34      collect2: error: ld returned 1 exit status
     35      /tmp/blosc2_list_compressors0ckok9tf.c: In function 'main':
     36      /tmp/blosc2_list_compressors0ckok9tf.c:2:5: warning: implicit declaration of function 'blosc2_list_co
           mpressors' [-Wimplicit-function-declaration]
     37          2 |     blosc2_list_compressors();
     38            |     ^~~~~~~~~~~~~~~~~~~~~~~
  >> 39      /usr/bin/ld: cannot find -lblosc2: No such file or directory
  >> 40      collect2: error: ld returned 1 exit status
```

**Comment by adamjstewart** [comment]
Thanks @manuelakuhn. If you have time, can you try to get those working? Even if I fix those I can't build with Apple Clang 15 so it's hard for me to debug until we get that fixed. If you don't have time, I can try to see how far I can get tomorrow.

**Comment by manuelakuhn** [comment]
I will try. So far I can tell that running `spack install py-tables+lzo+bzip2` gets rid of all errors except of the one for `-lblosc2`, so it seems that those are not optional dependencies.

File: var/spack/repos/builtin/packages/py-tables/package.py [hunk 2]
```diff
@@ -25,26 +26,35 @@ class PyTables(PythonPackage):
...
+
+    # Historical dependencies
+    depends_on("py-six@1.9:", when="@:3.5", type=("build", "run"))
+
+    # tables/req_versions.py
     # Versions prior to 3.3 must build with the internal blosc due to a lock
     # problem in a multithreaded environment.
-    depends_on("c-blosc@1.4.1:", when="@3.3.0:")
+    depends_on("c-blosc@1.11.1:", when="@3.8:")
+    depends_on("c-blosc@1.4.1:", when="@3.3:")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-tables/package.py:57]:
```
depends_on("c-blosc2@2.5.0:", when="@3.8:")
```
and further down
```
        if "^c-blosc2" in self.spec:
            env.set("BLOSC2_DIR", self.spec["c-blosc2"].prefix)
```
see [req_versions.py#L14](https://github.com/PyTables/PyTables/blob/v3.8.0/tables/req_versions.py#L14)

Sorry overlooked this before, but I still get the 
```
  .. ERROR:: Unable to find the blosc2 library.
```

**Comment by manuelakuhn** [comment]
I found this [issue](https://github.com/PyTables/PyTables/issues/998) which seems to have the same problem. The [patch]( https://github.com/PyTables/PyTables/pull/1000) they say we should use does not apply on `3.8.0`.
I did some tests with the master branch and this builds just fine (when `c-blosc2` is added to the spack recipe). According to their timeline in the `release/3.9.0` branch they plan to release `3.9.0` today. So maybe we should wait till this is out and work from there.

**Comment by manuelakuhn** [comment]
Release `3.9.0` installs fine even without an external `c-blosc2` dependency and also no variants needed. Will open a pull request.

**Pull request merged**